ansible_rulebook/websocket.py

```diff
@@ -199,6 +201,20 @@ async def _handle_request_workload(
             if not data.get("data") and not data.get("more"):
                 os.close(project_data_fh)
                 logger.debug("wrote %s", response.project_data_file)
+        if data.get("type") == "FileContents":
+            template_key = data.get("template_key")
+            raw_data = base64.b64decode(data.get("data"))
+            keys = template_key.split(".")
+            if len(keys) == 1 and template_key == "template":
+                key = "filename"
+                non_fq_key = True
+            else:
+                key = keys[1]
```

Comment:
Does this presuppose there is, at most, two fields separated by '.'?  If the template key was `one.two.three` this would selected `two` discarding `three`.

Reply:
@jshimkus-rh Its a quirk in the default keyname when there is only 1 key
```
{
  "file": {
      "template": "[mycloud]\ntoken={{ api_token }}"
  },
  "env": {
      "THIRD_PARTY_CLOUD_API_TOKEN": "{{ api_token }}"
  },
  "extra_vars": {
      "some_extra_var": "{{ username }}:{{ password }}"
  }
}```
From this document https://docs.ansible.com/automation-controller/latest/html/userguide/credential_types.html
It supports single template with a value of filename as well as template with different file names
In the example below the filename key would be cafile/keyfile/certfile since there is more than one.
```
file:
  template.cafile: '{{cafile}}'
  template.keyfile: '{{keyfile}}'
  template.certfile: '{{certfile}}'
```

Reply:
This is the sort of thing that drives me nuts.  If controller is going to define the format for this sort of thing it should be providing an api that decomposes the appropriate portion of a payload containing this info.  For specialization by other software (e.g., EDA) that api should be extensible for the specialization or at least include a way of providing what to look for in that specialization.

It doesn't matter how trivial the implementation of that api might be the functionality represented by the definition should be a complete solution.